### ISSUE TITLE
"EnforcedShorthandSyntax: consistent" doesn't work?

### ISSUE THREAD

**Issue by jdelStrother** [rubocop/rubocop#10933]
I'm using rubocop 1.35, which includes the `EnforcedShorthandSyntax: consistent` option in Style/HashSyntax (https://github.com/rubocop/rubocop/pull/10776)

However in my experience, `consistent` just behaves like `always`.

## Expected behavior

`EnforcedShorthandSyntax: consistent` should allow `{foo:, bar:}` and `{foo: foo, bar: bar}`

## Actual behavior

`EnforcedShorthandSyntax: consistent` only allows `{foo:, bar:}`

.rubocop.yml :

```yaml
AllCops:
  TargetRubyVersion: 3.1
  DisabledByDefault: true

Style/HashSyntax:
  Enabled: true
  EnforcedShorthandSyntax: consistent
```

test.rb :

```rb
foo, bar, baz = nil
short = { foo: }
long = { foo: foo }
short_consistent = { foo:, bar: }
long_consistent = { foo: foo, bar: bar}
inconsistent = { foo: foo, bar: }
```

```
$ bundle exec rubocop --debug
For /Users/jon/Developer/consistent-hash: configuration from /Users/jon/Developer/consistent-hash/.rubocop.yml
Default configuration from /Users/jon/.gem/ruby/3.1.0/gems/rubocop-1.35.0/config/default.yml
Use parallel by default.
Running parallel inspection
Loading cache from /Users/jon/.cache/rubocop_cache/71589f41d68f90ef315fa6058661f86c7fa52057/6d7a3b621ca1730e04accd938619e4bdab66cfb1/009fb4a3f5837ba88be0887547b8a871a0c297c1
Inspecting 2 files
Scanning /Users/jon/Developer/consistent-hash/Gemfile
Loading cache from /Users/jon/.cache/rubocop_cache/71589f41d68f90ef315fa6058661f86c7fa52057/6d7a3b621ca1730e04accd938619e4bdab66cfb1/009fb4a3f5837ba88be0887547b8a871a0c297c1
.Scanning /Users/jon/Developer/consistent-hash/test.rb
Loading cache from /Users/jon/.cache/rubocop_cache/71589f41d68f90ef315fa6058661f86c7fa52057/6d7a3b621ca1730e04accd938619e4bdab66cfb1/495fb8579da040f56bf4f00e9f5114634a98f29e
C

Offenses:

test.rb:3:15: C: [Correctable] Style/HashSyntax: Omit the hash value.
long = { foo: foo }
              ^^^
test.rb:5:26: C: [Correctable] Style/HashSyntax: Omit the hash value.
long_consistent = { foo: foo, bar: bar}
                         ^^^
test.rb:5:36: C: [Correctable] Style/HashSyntax: Omit the hash value.
long_consistent = { foo: foo, bar: bar}
                                   ^^^
test.rb:6:23: C: [Correctable] Style/HashSyntax: Do not mix explicit and implicit hash values. Omit the hash value.
inconsistent = { foo: foo, bar: }
                      ^^^

2 files inspected, 4 offenses detected, 4 offenses autocorrectable
Finished in 0.2000120000448078 seconds
```

I would have expected that I get only one offense reported here, on `inconsistent = { foo: foo, bar: }`. What am I missing @h-lame ?

## RuboCop version

```
$ bundle exec rubocop -V
1.35.0 (using Parser 3.1.2.1, rubocop-ast 1.21.0, running on ruby 3.1.2 arm64-darwin21)
```

**Comment by Darhazer** [comment]
The `consistent` style requires that all hash values are omitted, if that's possible, or that none of the values are omitted, if there are values that could not be omitted.
For instance:
```
inconsistent = { foo: fu, bar: }
```
would be reported as inconsistent, as it should be `{ foo: fu, bar: bar }`

I guess the documentation/naming is not clear in the case.
```
consistent - like "always", but will avoid mixing styles in a single hash
```
This means that it "forces use of the 3.1 syntax (e.g. {foo:})", but will avoid mixing styles; e.g. will disallow shorthand version if that would lead to mixing styles.
In your case using the shorthand style will not lead to mixing styles, that's why it's reported.  But that's why all long versions are reported as well.

**Comment by jdelStrother** [comment]
Oh I see - thanks, that makes more sense. I think I'll stick with `either` for now.
Happy to close this, unless you want to leave it open for documentation improvements.

**Comment by dvandersluis** [comment]
Gonna close it, thanks @jdelStrother!

**Comment by h-lame** [comment]
Hey @jdelStrother - sorry the description of `consistent` wasn't 100%.  I think the examples are, but the one-liner could be clearer for sure.  Would:

```
consistent - avoids mixing short and long styles in a single hash by forcing use of shorthand syntax but only if all the values can be omitted in the hash, otherwise forcing explicit values for all keys
```

be clearer?

**Comment by jdelStrother** [comment]
No problem. Now I re-read the example I think it's ok - 

```
# bad
{foo: , bar: bar}

# good
{foo:, bar:}

# bad
{foo: , bar: baz}

# good
{foo: foo, bar: baz}
```

I think I maybe missed the one-letter difference in bar & baz on the last line, and thought it was saying that `{foo: foo, bar: bar}` was good.

---

> Would:
> > consistent - avoids mixing short and long styles in a single hash by forcing use of shorthand syntax but only if all the values can be omitted in the hash, otherwise forcing explicit values for all keys
>
> be clearer?

Yes, although I think it could be briefer. Maybe something like this?

> consistent - enforce short style if all values can be omitted in the hash, otherwise enforce long style.
